xbmc/cores/VideoPlayer/VideoRenderers/WinRenderer.cpp

```diff
@@ -231,49 +231,40 @@ int CWinRenderer::NextYV12Texture()
     return -1;
 }
 
-bool CWinRenderer::AddVideoPicture(DVDVideoPicture* picture, int index)
+bool CWinRenderer::IsPictureHW(DVDVideoPicture &picture)
 {
-  if (!m_NumYV12Buffers)
-    return false;
-
-  int source = index;
-  if (source < 0 || NextYV12Texture() < 0)
-    return false;
-
-  if (m_renderMethod == RENDER_DXVA || picture->format == RENDER_FMT_DXVA)
+  if (m_renderMethod == RENDER_DXVA
+    || picture.format == RENDER_FMT_DXVA)
   {
-    if (m_renderMethod == RENDER_DXVA)
-    {
-      DXVABuffer *buf = reinterpret_cast<DXVABuffer*>(m_VideoBuffers[source]);
-      SAFE_RELEASE(buf->pic);
-
-      if (picture->format == RENDER_FMT_DXVA)
-      {
-        if (picture->dxva)
-          buf->pic = picture->dxva->Acquire();
-      }
-      else
-      {
-        buf->pic = m_processor->Convert(picture);
-      }
-      buf->frameIdx = m_frameIdx;
-      m_frameIdx += 2;
-      return true;
-    }
-    else if (picture->format == RENDER_FMT_DXVA)
-    {
-      YUVBuffer *buf = reinterpret_cast<YUVBuffer*>(m_VideoBuffers[source]);
-      if (buf->IsReadyToRender())
-        return false;
-
-      return buf->CopyFromDXVA(reinterpret_cast<ID3D11VideoDecoderOutputView*>(picture->dxva->view));
-    }
+    return true;
   }
   return false;
 }
 
+void CWinRenderer::AddVideoPictureHW(DVDVideoPicture &picture, int index)
+{
+  if (m_renderMethod == RENDER_DXVA)
+  {
+    DXVABuffer *buf = reinterpret_cast<DXVABuffer*>(m_VideoBuffers[index]);
+    SAFE_RELEASE(buf->pic);
+    buf->pic = m_processor->Convert(picture);
+    buf->frameIdx = m_frameIdx;
+    m_frameIdx += 2;
+  }
+  else if (picture.format == RENDER_FMT_DXVA)
```

Comment:
Is this path ever taken? doesn't RENDER_FMT_DXVA imply m_renderMethod == RENDER_DXVA?